### PR TITLE
scripts(ticdc): refine check diff line

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: ref #4780 

### What is changed and how it works?

* Fix check failure when there is no master branch
* Add color to WARN and ERROR

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by overvenus** [comment]
/unhold

Tested with pr/4771, see https://ci.pingcap.net/blue/organizations/jenkins/lint-check/detail/lint-check/2395/pipeline/

**Review by hi-rustin — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

File: scripts/check-diff-line-width.sh [hunk 1]
```diff
@@ -16,18 +16,22 @@
 
 set -e
 
-git --no-pager diff master -U0 cdc pkg |
+# The hash of the latest commit with a commit message matching
+# the pattern `\(#[0-9]+\)$`. It's usually be a master branch commit.
```
Comment (overvenus) [inline scripts/check-diff-line-width.sh:20]:
```suggestion
# the pattern `\(#[0-9]+\)$`. It's usually a master branch commit.
```

**Pull request merged**